onnxruntime/core/providers/cuda/test/greedy_search_top_one.cc

```diff
@@ -18,7 +18,7 @@ void FillAndShuffle(std::vector<float>& values, int32_t batch_size, int32_t voca
   std::mt19937 generator(rd());
```

Comment:
Can we add an assert here that we do never exceed the values size since the batch_size and vocab_size seem to be disjoint from the values.size()?

Reply:
This function is used in only one place (in line 48 of this file). It is like a private funciton, the usage is fine.
```
  int32_t batch_x_vocab = batch_size * vocab_size;
  std::vector<float> values(batch_x_vocab);
  FillAndShuffle(values, batch_size, vocab_size);
```
If it is a public function, we can add more checks.